### ISSUE TITLE
TestHelper issue when validating immutable objects

### ISSUE THREAD

**Issue by JamesFenton** [JeremySkinner/FluentValidation#392]
When validating an object that has properties with no setter, for example
```
public class Command
    {
        public string Name { get; }

        public Command(string name)
        {
            Name = name;
        }
    }
```
with a validator
```
public class Validator : AbstractValidator<Command>
    {
        public Validator()
        {
            RuleFor(c => c.Name).NotEmpty()
        }
    }
```
Calling the following test helper method
```
var validator = new Validator();
var command = new Command("");
validator.ShouldHaveValidationErrorFor(c => c.Name, command);
```
throws the following exception
```
Expression must be writeable
Parameter name: left
   at System.Linq.Expressions.Expression.RequiresCanWrite(Expression expression, String paramName)
   at System.Linq.Expressions.Expression.Assign(Expression left, Expression right)
   at FluentValidation.MemberAccessor`2.CreateSetExpression(Expression`1 getExpression)
   at FluentValidation.MemberAccessor`2..ctor(Expression`1 getExpression)
   at FluentValidation.MemberAccessor`2.op_Implicit(Expression`1 this)
   at FluentValidation.TestHelper.ValidationTestExtension.TestValidate[T,TValue](IValidator`1 validator, Expression`1 expression, T instanceToValidate, TValue value, String ruleSet)
   at FluentValidation.TestHelper.ValidationTestExtension.ShouldHaveValidationErrorFor[T,TValue](IValidator`1 validator, Expression`1 expression, T objectToTest, String ruleSet)
   at MyTestMethod()
```
It seems that the property is trying to be set, but it shouldn't need to be if you pass in the object being validated. Is this to be expected, or is something unexpected happening? The exception is not thrown when the property has a private setter.

**Comment by JeremySkinner** [comment]
Internally the TestHelper always compiles setters, even if it doesn't end up using them.....I think my original version worked ok for this, but it's been rewritten by various people a couple of times and I think that subtlety was lost. I can look at re-adding support for this in a future release, but for now I'd suggest performing your assertions manually rather than use the testing extensions.

**Comment by JeremySkinner** [comment]
This is implemented ready for the 7.4 release. Sorry it's nearly a year late!

**Comment by JeremySkinner** [comment]
I've pushed 7.4.0 to nuget which contains this fix.
